### ISSUE TITLE
download_images no longer working

### ISSUE THREAD

**Issue by kevinbird15** [fastai/fastai#3445]
**Describe the bug**
[This commit](https://github.com/fastai/fastai/commit/63b624c696c956a850cc137878d2582402c1db92) removed download_url and didn't provide anything to replace it.  This function was being used by https://github.com/fastai/fastai/blob/dc3b9ad5242f00bfcc164b15c5ad1b666a8401d8/fastai/vision/utils.py#L29 and is therefore failing gracefully when download_images is being called downstream.  

**To Reproduce**
```
from fastai.vision.all import *
download_images('test', urls=['https://www.fast.ai/images/jh-head.jpg'])
assert (Path(f'test/00000001.jpg').is_file())
```

**Expected behavior**
If the image exists at the provided url, the image should be downloaded 

**Additional context**
This is causing issues in fastbook chapter 2, but would also cause issues anywhere that somebody is using fastai's download_images function so it is a wider issue than just fastbook.  I believe adding a function back into data.external to replace the download_url that is backwards compatible would be the best thing to do.

**Comment by finale80** [comment]
@kevinbird15 I noticed exactly the same

However, I think it's just code refactoring, and (I think) some glitches in the packaging

Looking at the code, `download_url` is not moved to `fastdownload` which is imported in the module 
https://github.com/fastai/fastai/blob/dfc4da0cac24df3b1fa0583d8bd8f9cba4ba874a/fastai/vision/utils.py

According to the track, @jph00 fixed it yesterday
https://github.com/fastai/fastai/commit/dfc4da0cac24df3b1fa0583d8bd8f9cba4ba874a

But I think the package is not available yet
Installing from the git itself it works for me
```
pip install git+https://github.com/fastai/fastai
```

@kevinbird15 your test seems downloading a file `00000000.jpg` not `00000001.jpg` so please double check your assert

**Comment by finale80** [comment]
Beside the above, I still find the try/except of `download_url` odd

```
try: download_url(url, dest/f"{name}{suffix}", show_progress=False, timeout=timeout)
except Exception as e: f"Couldn't download {url}."
```

the `except` leads to creating just a string...which it's not used :D
I assume there is a print missing?
Or re-raise the exception with the new message?

**Comment by matteopilotto** [comment]
@finale80 you saved me. Thanks!!!

**Comment by dhruvmetha** [comment]
Hey @kevinbird15 

The issue is that there is an extra parameter "overwrite" in the download_url function call inside fastai.vision.utils._download_image_inner

If we were to remove this, the function download_images in chapter 2 notebook would run smoothly

**Comment by jph00** [comment]
This should be fixed in v2.5.1 which is now available from pip and conda.
